extensions/ql-vscode/src/variant-analysis/variant-analysis-manager.ts

```diff
@@ -613,12 +617,33 @@ export class VariantAnalysisManager
             });
           }
         };
-        await this.variantAnalysisResultsManager.download(
-          variantAnalysis.id,
-          repoTask,
-          this.getVariantAnalysisStorageLocation(variantAnalysis.id),
-          updateRepoStateCallback,
-        );
+        let retry = 0;
+        for (;;) {
```

Comment:
Could this perhaps contain the `if` condition for the number of retries? That would make it somewhat easier to see that this is using retries rather than being an infinite loop.

Reply:
I was unable to avoid `while(true)` because it needs to rethrow if max attempts were reached instead of continuing the loop and ending it on some condition. Of course it is possible to write a "fake" `while (retry < maxRetries)` which on one hand duplicates the maxRetries check and on the other hand won't ever be reached.
If you can come up how to rewrite it I'm open for suggestions.

Reply:
Makes sense, thanks for thinking about it